subsys/sdfw_services/services/device_info/device_info_service.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/** .. include_startingpoint_echo_source_rst */
+#include <stddef.h>
+#include <stdint.h>
+#include <string.h>
+
+#include <sdfw/sdfw_services/device_info_service.h>
+
+#include "device_info_service_decode.h"
+#include "device_info_service_encode.h"
+#include "device_info_service_types.h"
+#include <sdfw/sdfw_services/ssf_client.h>
+#include "ssf_client_os.h"
+
+SSF_CLIENT_LOG_REGISTER(device_info_service, CONFIG_SSF_DEVICE_INFO_SERVICE_LOG_LEVEL);
+
+SSF_CLIENT_SERVICE_DEFINE(device_info_srvc, DEVICE_INFO, cbor_encode_device_info_req,
+			  cbor_decode_device_info_resp);
+
+enum {
+	UUID = 0,
+	HWVERSION,
+} device_info_target;
+
+enum {
+	SUCCESS = 0,
+	INTERNAL_ERROR = 0x1001001,
+	UNPROGRAMMED = 0x1001002,
```

Comment:
Is it okay to reuse `0x100_1001` and `0x100_1002` values which are already in use by IMEI service? Or vice versa, do we always want to use the same codes for these?

Reply:
this is now generated in an enum based on the cddl data model file.